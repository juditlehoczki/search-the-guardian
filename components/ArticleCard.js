import React from "react";
import {
  View,
  Text,
  Linking,
  Image,
  Row,
  TouchableOpacity,
} from "react-native";
import moment from "moment";
import s from "../styles.js";

const Card = (props) => {
  const { webTitle, webPublicationDate, sectionName, webUrl } = props.result;
  const { thumbnail, trailText } = props.result.fields;

  return (
    <TouchableOpacity
      onPress={() => {
        Linking.openURL(webUrl);
      }}
    >
      <View style={s.card}>
        <Image
          style={s.thumbnail}
          source={{
            uri: thumbnail,
          }}
        />
        <View style={s.intro}>
          <Text style={s.articleTitle}>{`${webTitle}`}</Text>
          <View style={{ flexDirection: "row" }}>
            <Text>{`${moment(webPublicationDate).format("DD-MM-YYYY")}`}</Text>
            <Text> in</Text>
            <Text style={s.date}> {sectionName}</Text>
          </View>
          <Text>{`${trailText}...`}</Text>
        </View>
        <Text style={s.more}>&gt;</Text>
      </View>
    </TouchableOpacity>
  );
};

export default Card;
