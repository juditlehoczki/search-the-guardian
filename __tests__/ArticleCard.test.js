import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {TouchableOpacity, Linking} from 'react-native';

import ArticleCard from '../components/ArticleCard.js';

const props = {
  result: {
    webTitle: '',
    webPublicationDate: '',
    sectionName: '',
    webUrl: '',
    fields: {
      thumbnail: '',
      trailText: '',
    },
  },
};

describe('ArticleCard Component', () => {
  test('Snapshot - it renders correctly', () => {
    const snapshot = renderer.create(<ArticleCard {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('Tapping on the ArticleCard calls Linking.openURL function', () => {
    const fakeArticleCard = renderer.create(<ArticleCard {...props} />);
    const article = fakeArticleCard.root.findByType(TouchableOpacity);
    article.props.onPress();
    expect(Linking.openURL).toHaveBeenCalled();
  });
});
