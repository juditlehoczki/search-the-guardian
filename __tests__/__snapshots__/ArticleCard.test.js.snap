// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard Component Snapshot - it renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "marginHorizontal": 5,
        "marginVertical": 3,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "borderColor": "#54C1B7",
          "borderRadius": 5,
          "borderWidth": 3,
          "flex": 2,
          "height": "100%",
          "marginBottom": "auto",
          "marginTop": "auto",
          "minHeight": 200,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 3,
          "marginHorizontal": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0F406E",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text>
          Invalid date
        </Text>
        <Text>
           in
        </Text>
        <Text
          style={
            Object {
              "color": "#D1529E",
              "fontWeight": "bold",
            }
          }
        >
           
          
        </Text>
      </View>
      <Text>
        ...
      </Text>
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#D1529E",
          "fontSize": 30,
          "textAlign": "right",
        }
      }
    >
      &gt;
    </Text>
  </View>
</View>
`;
